### ISSUE TITLE
cargo new creates additional git floder

### ISSUE THREAD

**Issue by allexgray** [rust-lang/cargo#10529]
### Problem

Folder Tree

- .git
- Cargo.toml

### After executing
```cargo new hello``` & ```cargo new app``` 

Folder Tree 

- .git
- app
  - src
  - Cargo.toml
- hello
  - .git
  - src
  - Cargo.toml
- Cargo.toml

#### Root Cargo.toml file
```toml
[workspace]
resolver = "2"
```

This happens once in every folder after executing ```cargo new hello``` & then ```cargo new app```  git was instantiated only in hello (first package of every directory) running ```cargo new package/hello``` will also produce same result.

### Steps

```
cargo new hello
cargo new app
``` 

### Possible Solution(s)

check if root folder is a git repository

### Notes

this only happens when you create first package in workspace followed by every new directory

### Version

```text
cargo 1.59.0 (49d8809dc 2022-02-10)
release: 1.59.0
commit-hash: 49d8809dc2d3e6e0d5ec634fcf26d8e2aab67130
commit-date: 2022-02-10
host: x86_64-pc-windows-msvc
libgit2: 1.3.0 (sys:0.13.23 vendored)
libcurl: 7.80.0-DEV (sys:0.4.51+curl-7.80.0 vendored ssl:Schannel)
os: Windows 10.0.19044 (Windows 10 Home Single Language) [64-bit]
```

**Comment by epage** [comment]
Let me rephrase the reproduction steps to make sure I understand.

In an empty directory, run
```
cargo new hello
cargo new app
```

Now, instead of having a directory with `hello/` and `app/`, you also have `Cargo.toml` and `.git/`

Is that correct?

On the same version of cargo but on Linux, I'm not able to reproduce.  I'll try to get my windows box up and reproduce it there.

**Comment by allexgray** [comment]
### Steps

create an empty directory app 

![Screenshot 2022-04-02 045843](https://user-images.githubusercontent.com/100816942/161365926-65a75bea-8704-406b-8781-2b8504abd683.png)

then executing ``` git init ```

![Screenshot 2022-04-02 050000](https://user-images.githubusercontent.com/100816942/161365940-1d8ca6fd-f4ac-4632-b49b-058559852b67.png)

![Screenshot 2022-04-02 050101](https://user-images.githubusercontent.com/100816942/161365950-b9c53bbe-4f9e-49f5-9f72-90fa2aaa8a8c.png)

creating Cargo.toml file into same directory 

![Screenshot 2022-04-02 050424](https://user-images.githubusercontent.com/100816942/161365962-ffb3cf3b-3c59-415c-91e4-4fc789a22397.png)

then executing ``` cargo new hello``` will create

app/hello: directory
app/hello/Cargo.toml: file
app/hello/src: directory
app/hello/src/main.rs: file
app/hello/.git: directory
app/hello/.gitignore: file

![Screenshot 2022-04-02 050702](https://user-images.githubusercontent.com/100816942/161365976-405cd107-bf47-40db-93ff-b57fcdcc03d7.png)

![Screenshot 2022-04-02 051038](https://user-images.githubusercontent.com/100816942/161365980-df24fc54-564c-4770-b1b4-b2eb5271eb73.png)

now because of that .git directory git treats hello as a separate repository you can see this when i execute ``` git add * ```

![Screenshot 2022-04-02 051129](https://user-images.githubusercontent.com/100816942/161366046-a628cf91-ba53-49d9-812c-bed99ed21ce7.png)

I can just delete the app/hello/.git directory to get it working 

![Screenshot 2022-04-02 051201](https://user-images.githubusercontent.com/100816942/161366139-29aa4c42-4738-4a97-8d72-a1b2065637e1.png)

![Screenshot 2022-04-02 051304](https://user-images.githubusercontent.com/100816942/161366180-cf410d77-e9b2-48c9-8ab3-e3ddd2a2615c.png)

but the .git folder is hidden by default in many ide and file manager it took me long time to notice why i was unable to commit my changes

**Comment by tasogare3710** [comment]
It does not reproduce in my environment. no .git directories will be created for both app and hello.
Could it be a issue in some versions?

- - -
My cargo

cargo 1.61.0-nightly (1ef1e0a12 2022-03-31)
release: 1.61.0-nightly
commit-hash: 1ef1e0a12723ce9548d7da2b63119de9002bead8
commit-date: 2022-03-31
host: x86_64-pc-windows-msvc
libgit2: 1.4.2 (sys:0.14.2 vendored)
libcurl: 7.80.0-DEV (sys:0.4.51+curl-7.80.0 vendored ssl:Schannel)
os: Windows 10.0.19044 (Windows 10 Pro) [64-bit]
```

**Comment by allexgray** [comment]
fresh install of rust fixed this issues #10529

**Comment by Eh2406** [comment]
That's very odd, but I'm glad you got it figured out!
